src/App.js

```diff
@@ -26,43 +25,76 @@ const generateSquares = () => {
 }
 
 const App = () => {
-
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState('x');
+  const [winner, setWinner] = useState(null)
+  let newBoard = [...squares]
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
-
+  const updateSquare = (squareToUpdate) => {
+    const row = Math.floor(squareToUpdate.id/3)
+    const colume = squareToUpdate.id%3
 
+    if (squareToUpdate.value === '') {
+      squareToUpdate.value = currentPlayer
+      newBoard[row][colume] = squareToUpdate
+      if (currentPlayer === 'x') {
```

Comment:
This would be clearer if you used the constants:

```suggestion
      if (currentPlayer === PLAYER_1) {
```